modules/lang-painless/build.gradle

```diff
@@ -46,14 +49,39 @@ testClusters.all {
 
 dependencies {
   api 'org.antlr:antlr4-runtime:4.9.3'
-  api 'org.ow2.asm:asm-util:9.2'
-  api 'org.ow2.asm:asm-tree:9.2'
-  api 'org.ow2.asm:asm-commons:9.2'
-  api 'org.ow2.asm:asm-analysis:7.2'
-  api 'org.ow2.asm:asm:9.2'
+  implementation 'org.ow2.asm:asm-util:9.2'
```

Comment:
Any chance someone was relying on this dependency and using through lang-painless module?
```
#### implementation

Dependencies that are used by the project at compile and runtime but are not exposed as a compile dependency to other dependent projects. Dependencies added to the `implementation` configuration are considered an implementation detail that can be changed at a later date without affecting any dependent projects.

#### api

Dependencies that are used as compile and runtime dependencies of a project and are considered part of the external api of the project.
```

Reply:
Its possible, will revisit this if performing our own shadowing is the right path fwd, I'm not convinced thats the case.

Reply:
I've changed these back to api, but I don't think this will make a difference because we are rewriting the package namespace.